homeassistant/components/volvooncall/__init__.py

```diff
@@ -184,30 +119,21 @@ def discover_vehicle(self, vehicle):
         self.vehicles.add(vehicle.vin)
 
         dashboard = vehicle.dashboard(
-            mutable=self.config[DOMAIN][CONF_MUTABLE],
-            scandinavian_miles=self.config[DOMAIN][CONF_SCANDINAVIAN_MILES],
+            mutable=self.config_entry.data[CONF_MUTABLE],
+            scandinavian_miles=self.config_entry.data[CONF_SCANDINAVIAN_MILES],
         )
 
         for instrument in (
             instrument
             for instrument in dashboard.instruments
             if instrument.component in PLATFORMS
         ):
-
             self.instruments.add(instrument)
 
+        for platform in PLATFORMS:
             self.hass.async_create_task(
-                discovery.async_load_platform(
-                    self.hass,
-                    PLATFORMS[instrument.component],
-                    DOMAIN,
-                    (
-                        vehicle.vin,
-                        instrument.component,
-                        instrument.attr,
-                        instrument.slug_attr,
-                    ),
-                    self.config,
+                self.hass.config_entries.async_forward_entry_setup(
```

Comment:
Please use `hass.config_entries.async_forward_entry_setups` instead.

Reply:
We can't forward the entry to the platforms more than once. We need to add a discovery function to the platforms that can add new entities after the platform has been set up, and make sure that all platforms have been set up once before we signal the discovery function to add new entities. We can forward the entry to all the platforms before we discovery any vehicle.

Example:
https://github.com/home-assistant/core/blob/4239757c2c3a22bc2518211c69e58a8117a3e022/homeassistant/components/tuya/binary_sensor.py#L343-L371

https://github.com/home-assistant/core/blob/4239757c2c3a22bc2518211c69e58a8117a3e022/homeassistant/components/tuya/__init__.py#L262-L268